kratos/geometries/brep_point.h

```diff
@@ -0,0 +1,352 @@
+    typedef typename BaseType::GeometriesArrayType GeometriesArrayType;
+    typedef typename BaseType::IntegrationPointsArrayType IntegrationPointsArrayType;
+
+    static constexpr IndexType BACKGROUND_GEOMETRY_INDEX = -1;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for untrimmed patch
+    BrepPoint(
+        const CoordinatesArrayType LocalCoordinates,
+        typename GeometryType::Pointer pBackgroundGeometry)
+        : BaseType(PointsArrayType(), &msGeometryData)
+        , mLocalCoordinates(LocalCoordinates)
+        , mpBackgroundGeometry(pBackgroundGeometry)
+    {
+        KRATOS_ERROR_IF(pBackgroundGeometry->LocalSpaceDimension() != this->LocalSpaceDimension())
+            << "Local space dimension of background geometry (" << pBackgroundGeometry->LocalSpaceDimension()
+            << ") does not coincide with LocalSpaceDimension of this geometry (" << this->LocalSpaceDimension() << ")."
+            << std::endl;
+    }
+
+    explicit BrepPoint(const PointsArrayType& ThisPoints)
+        : BaseType(ThisPoints, &msGeometryData)
+    {
+    }
+
+    /// Copy constructor.
+    BrepPoint(BrepPoint const& rOther )
+        : BaseType( rOther )
+        , mLocalCoordinates(rOther.mLocalCoordinates)
+        , mpBackgroundGeometry(rOther.mpBackgroundGeometry)
+    {
+    }
+
+    /// Destructor
+    ~BrepPoint() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    BrepPoint& operator=( const BrepPoint& rOther )
+    {
+        BaseType::operator=( rOther );
+        mLocalCoordinates = rOther.mLocalCoordinates;
+        mpBackgroundGeometry = rOther.mpBackgroundGeometry;
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename BaseType::Pointer Create( PointsArrayType const& ThisPoints ) const override
+    {
+        return typename BaseType::Pointer( new BrepPoint( ThisPoints ) );
+    }
+
+    ///@}
+    ///@name Access to Geometry Parts
+    ///@{
+
+    /**
+    * @brief This function returns the pointer of the background geometry
+    *        of this point. Accesseable via BACKGROUND_GEOMETRY_INDEX.
+    * @param Index: only possible with BACKGROUND_GEOMETRY_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(IndexType Index) override
+    {
+        const auto& const_this = *this;
+        return std::const_pointer_cast<GeometryType>(
+            const_this.pGetGeometryPart(Index));
+    }
+
+    /**
+    * @brief This function returns the pointer of the background geometry
+    *        of this point. Accesseable via BACKGROUND_GEOMETRY_INDEX.
+    * @param Index: only possible with BACKGROUND_GEOMETRY_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    const GeometryPointer pGetGeometryPart(IndexType Index) const override
+    {
+        if (Index == BACKGROUND_GEOMETRY_INDEX)
```

Comment:
I do the check to avoid that users are trying to access different information, which cannot be accessed at this geometry. Considering, -1 is always the background geometry/ background surface/ curve which actually contains the degrees of freedoms. All numbers bigger than one are user defined geometries, as e.g. embedded objects. Accordingly, keeping this check would be good. It isn't called many times anyways.